packages/client/src/apis/market/MarketExecuteAPI.ts

```diff
@@ -23,7 +23,7 @@ export class MarketExecuteAPI extends BaseVertexAPI {
     const sender = await this.getChainSignerAddress();
 
     return this.context.engineClient.mintLp({
-      verifyingAddr: this.context.contracts.endpoint.address,
+      verifyingAddr: await this.getEndpointAddress(),
```

Comment:
what if we make the endpoint address a property on the client somewhere and only fetch it once on instantiation?

Reply:
ye, i think we should do that 👍